azurecontainerapps/azurecontainerapps.ts

```diff
@@ -30,200 +23,487 @@ export class azurecontainerapps {
+                    this.targetPort = '80';
+                } else {
+                    this.targetPort = '8080';
+                }
+
+                console.log(tl.loc('DefaultTargetPortMessage', this.targetPort));
             }
 
-            // If a Dockerfile was found or provided, create a runnable application image from that
-            if (!util.isNullOrEmpty(dockerfilePath) && shouldBuildAndPushImage) {
-                console.log(tl.loc('CreateImageWithDockerfileMessage', dockerfilePath));
-                appHelper.createRunnableAppImageFromDockerfile(imageToDeploy, appSourcePath, dockerfilePath);
+            // Set the target port to 80 if it was not provided or determined
+            if (util.isNullOrEmpty(this.targetPort)) {
+                this.targetPort = '80';
+                console.log(tl.loc('DefaultTargetPortMessage', this.targetPort));
+            }
 
-                // If telemetry is enabled, log that the Dockerfile scenario was targeted for this task
-                telemetryHelper.setDockerfileScenario();
+            // Add the ingress value and target port to the optional arguments array
+            // Note: this step should be skipped if we're updating an existing Container App (ingress is enabled via a separate command)
+            if (!this.containerAppExists) {
+                this.commandLineArgs.push(`--ingress ${this.ingress}`);
+                this.commandLineArgs.push(`--target-port ${this.targetPort}`);
             }
+        }
 
-            // Push image to Azure Container Registry
-            if (shouldBuildAndPushImage) {
-                new ContainerRegistryHelper().pushImageToAcr(imageToDeploy);
+        const environmentVariables: string = tl.getInput('environmentVariables', false);
+
+        // Add user-specified environment variables
+        if (!util.isNullOrEmpty(environmentVariables)) {
+            if (this.containerAppExists) {
+                this.commandLineArgs.push(`--replace-env-vars ${environmentVariables}`);
             } else {
-                // If telemetry is enabled, log that the previously built image scenario was targeted for this task
-                telemetryHelper.setImageScenario();
+                this.commandLineArgs.push(`--env-vars ${environmentVariables}`);
             }
+        }
+    }
 
-            // Create or update Azure Container App
-            appHelper.createOrUpdateContainerApp(containerAppName, resourceGroup, imageToDeploy, optionalCmdArgs);
-            telemetryHelper.setSuccessfulResult();
-        } catch (err) {
-            tl.setResult(tl.TaskResult.Failed, err.message);
-            telemetryHelper.setFailedResult(err.message);
-        } finally {
-            // Logout of Azure if logged in during this task session
-            authHelper.logoutAzure();
+    /**
+     * Creates or updates the Container App.
+     */
+    private static createOrUpdateContainerApp() {
```

Comment:
It might be a matter of personal preference, so please feel free to ignore if you feel strongly. Please consider returning early in a couple of cases below to reduce nesting, such as:
```
if (!this.containerAppExists) {
    if (!util.isNullOrEmpty(this.yamlConfigPath)) {
        // Create the Container App from the YAML configuration file
        this.appHelper.createContainerAppFromYaml(this.containerAppName, this.resourceGroup, this.yamlConfigPath);
    } else {
        // Create the Container App from command line arguments
        this.appHelper.createContainerApp(this.containerAppName, this.resourceGroup, this.containerAppEnvironment, this.imageToDeploy, this.commandLineArgs);
    }
    return;
}

if (!util.isNullOrEmpty(this.yamlConfigPath)) {
    // Update the Container App from the YAML configuration file
    this.appHelper.updateContainerAppFromYaml(this.containerAppName, this.resourceGroup, this.yamlConfigPath);
    return;
}

// Update the Container App from command line arguments
this.appHelper.updateContainerApp(this.containerAppName, this.resourceGroup, this.imageToDeploy, this.commandLineArgs);

// Update ingress on the Container App
if (this.ingressEnabled) {
    this.appHelper.enableContainerAppIngress(this.containerAppName, this.resourceGroup, this.targetPort, this.ingress);
} else {
    this.appHelper.disableContainerAppIngress(this.containerAppName, this.resourceGroup);
}

// Update the ACR details if provided
if (!util.isNullOrEmpty(this.acrName) && !util.isNullOrEmpty(this.acrUsername) && !util.isNullOrEmpty(this.acrPassword)) {
    this.appHelper.updateContainerAppRegistryDetails(this.containerAppName, this.resourceGroup, this.acrName, this.acrUsername, this.acrPassword);
}
```